lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -101,13 +165,39 @@ class tethys::ScienceSensorsSystemPrivate
   public: const std::string NORTH_CURRENT {
     "northward_sea_water_velocity_meter_per_sec"};
 
+  ////////////////////////////
+  // Fields for bookkeeping
+
   /// \brief Input data file name, relative to a path Ignition can find in its
   /// environment variables.
   public: std::string dataPath {"2003080103_mb_l3_las.csv"};
 
+  /// \brief Coordinates where sensor location was last interpolated.
+  /// Helps to determine whether sensor location needs to be updated
+  public: ignition::math::Vector3d lastSensorPosENU = {
+    std::numeric_limits<double>::max(),
+    std::numeric_limits<double>::max(),
+    std::numeric_limits<double>::max()};
+
+  /// \brief Distance robot needs to move before another data interpolation
+  /// (based on sensor location) takes place.
+  public: const float INTERPOLATE_DIST_THRESH = 5.0;
```

Comment:
```suggestion
  // TODO(tfoote) Avoid this magic number see below for more context.
  private: const float INTERPOLATE_DIST_THRESH = 5.0;
```

Since this is a magic number that's unadjustable, lets suppress it away from the public API.

Reply:
Just a note that this is Pimplized. It is a private class, so everything in the class is already private. Just Ignition standard style - private class, all members are public.